pkg/api/testapi/testapi.go

```diff
@@ -24,63 +24,127 @@ import (
 
 	"k8s.io/kubernetes/pkg/api/latest"
 	"k8s.io/kubernetes/pkg/api/meta"
+	apiutil "k8s.io/kubernetes/pkg/api/util"
+	explatest "k8s.io/kubernetes/pkg/expapi/latest"
 	"k8s.io/kubernetes/pkg/runtime"
 )
 
-// Version returns the API version to test against, as set by the KUBE_API_VERSION env var.
-func Version() string {
-	version := os.Getenv("KUBE_API_VERSION")
-	if version == "" {
-		version = latest.Version
+// GroupAndVersion returns the API version to test against for a group, as set
+// by the KUBE_TEST_API env var.
+// Return value is in the form of "group/version".
+// If there is no group input, return "v1".
+func GroupAndVersion(group ...string) string {
+	if len(group) == 0 {
+		return "v1"
 	}
-	return version
+
+	version := os.Getenv("KUBE_TEST_API")
+	testGroupVersions := strings.Split(version, ",")
+
+	for _, groupVersion := range testGroupVersions {
+		if apiutil.GetGroup(groupVersion) == group[0] {
+			return groupVersion
+		}
+	}
+	//TODO: we need a central place to store all available API groups and their metadata in their respective latest.go
+	if group[0] == "experimental" {
+		return explatest.Version
+	} else {
+		panic(fmt.Errorf("cannot find appropriate testing version for group %s", group))
+	}
+}
+
+// Version implements the same logic as GroupAndVersion but the return value is
+// in the form of "version"
```

Comment:
Document that the ... parameter is to allow you to omit the parameter (is that actually a good idea?), not to accept multiple parameters.